### PR TITLE
splitted responsibilities

### PR DESCRIPTION
**Pull request by mensfeld**
close https://github.com/karafka/karafka/issues/463

cc: @benissimo

note: does not provide way to overwrite the default serializer in the responder as this will be provided in a different PR.

### CODE REVIEW

File: lib/karafka/base_responder.rb [hunk 1]
```diff
@@ -82,16 +82,18 @@ class << self
       # @param topic_name [Symbol, String] name of topic to which we want to respond
       # @param options [Hash] hash with optional configuration details
       def topic(topic_name, options = {})
+        options[:serializer] ||= Karafka::App.config.serializer
+        options[:registered] = true
```
Comment (benissimo) [inline lib/karafka/base_responder.rb]: This makes the `options.merge(registered: true)` below redundant.

Comment (benissimo) [reply lib/karafka/base_responder.rb]: @mensfeld ^ this is my only feedback.

Comment (mensfeld) [reply lib/karafka/base_responder.rb]: True! Thanks!

File: lib/karafka/params/params.rb [hunk 2]
```diff
@@ -40,30 +40,28 @@ class Params < Hash
...
-          self['parser'].parse(payload)
+      # @param payload [String] Raw data that we want to deserialize using consumer deserializer
+      # @return [Object] deserialized data
+      def deserialize(payload)
+        Karafka.monitor.instrument('params.params.deserialize', caller: self) do
+          self['deserializer'].call(payload)
         end
-      rescue ::Karafka::Errors::ParserError => e
-        Karafka.monitor.instrument('params.params.parse.error', caller: self, error: e)
+      rescue ::StandardError => e
```
Comment (nijikon) [inline lib/karafka/params/params.rb]: I'm not sure about this, can't we catch a more specific error here?

Comment (mensfeld) [reply lib/karafka/params/params.rb]: The question is, should require users to raise internal karafka error upon their deserializer execution or should we just allow it to raise whatever it wants and handle that. I would go with the second as it is less problematic for the end user, plus anyhow we don't have the "raw" data injection anymore from at leats `1.1`, so we reraise after the instrumentation.

Comment (nijikon) [reply lib/karafka/params/params.rb]: Sounds good

**Pull request merged**